architecture.md

```diff
@@ -0,0 +1,91 @@
+
+The Ignition Physics library is only used in the physics plugin, not in other plugins nor in the core of Ignition Gazebo.
+Some libraries are only used by one plugin, or one in each process (frontend and backend).
+However, some foundational libraries are used by all plugins, like [Ignition Common](https://ignitionrobotics.org/libs/common) which provides common functionality like logging, manipulating strings, file system interaction, etc., to all plugins.
+Other such libraries include [Ignition Plugin](https://ignitionrobotics.org/libs/plugin), [Ignition Math](https://ignitionrobotics.org/libs/math), [SDFormat](https://ignitionrobotics.org/libs/sdformat), and more.
+
+There are certain plugins in both the frontend and backend processes that are loaded by default in every version of Ignition.
+Many other plugins are optional.
+One common optional plugin is the sensors plugin.
+Both optional and default plugins can be removed or added at any time; Ignition Gazebo will continue to run with limited functionality.
+These demos on [Server Configuration](https://ignitionrobotics.org/api/gazebo/4.3/server_config.html) and [GUI Configuration](https://ignitionrobotics.org/api/gazebo/4.3/gui_config.html) showcase that functionality.
+
+The simulation process is depicted in the diagram below, and further explained in the Backend and Frontend process sections that follow.
+
+![Ignition Gazebo architecture diagram](images/IgnitonGazeboArchicteture.png)
+
+## Backend server process
+
+Ignition Gazebo is responsible for loading plugins in the backend, referred to as systems.
+The server runs an entity-component system architecture (see [Ignition Gazebo terminology](https://ignitionrobotics.org/api/gazebo/4.2/terminology.html)).
+The backend will usually have multiple systems responsible for everything in the simulation – computing physics, recording logs, receiving user commands, etc.
+
+Systems act on entities and components of those entities.
+An entity is anything in the simulation scene (a model, link, light, actor, etc.), and components are their characteristics (pose, name, geometry, etc.).
+Server plugins have access to all entities and what components they have, and make modifications to those components.
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Other Ignition Libraries, however, don’t know about the entities and components.
+
+There is a loop running in the backend that runs the systems, where some systems are proposing changes to entities and their components, and other systems are handling and applying those changes.
+This is called the “simulation loop”.
```

Comment:
I believe the "simulation loop" consists of `PreUpdate`, `Update`, and `PostUpdate`, right? Perhaps we can talk about this here, unless that is a little too technical/detailed for the scope of this tutorial. @chapulina what do you think?

Reply:
@maryaB-osr and I briefly discussed this idea but we thought this may be a good topic for a follow-up tutorial, not to make this one too long.

Reply:
Doing this in a follow-up tutorial is fine with me. I'll mark this as resolved.